ssl/ssl_local.h

```diff
@@ -1226,8 +1226,8 @@ typedef struct cert_pkey_st CERT_PKEY;
 struct quic_data_st {
     struct quic_data_st *next;
     OSSL_ENCRYPTION_LEVEL level;
-    size_t start;       /* offset into quic_buf->data */
     size_t length;
+    /* char data[]; should be here but C90 VLAs not allowed here */
```

Comment:
Paragraph  6.7.3.2.20 of n3220 permits flexible arrays as last element which would be allowed. However I think we need to double check what century our compilers are from. This is probably the clean way to do what you want, as I'm not sure if what you're doing is undefined behavior or not in all places

Reply:
I would also be inclined to use real C99 flexible arrays.
On a different project investigating the question of flexible array support, my colleague wrote "Using the godbolt.org side, I couldn't find a gcc or clang level that didn't support flex arrays.".  That project ended up using an autoconf test with some probably-undefined-behavior alternative for systems that don't support flexible arrays (e.g., AIX), whose compilers can probably safely be assumed to not be trying to do anything with that undefined behavior.
quictls inherits its configure script from openssl, and does not have real facility for autoconf-style "try a build and see if it works" checks, but our target audience is also probably for modern enough systems that we could just assume support for it.  (That same other project I mentioned also concluded that Microsoft's compilers supported it as an extension well before it was standardized.)

Reply:
I don't have any strong preference.  That part of the change was simply restoring code from https://github.com/quictls/openssl/commit/075ede1dd8229425e494194188f0629857aeb52d.

Reply:
I'm good with flexible arrays. @nibanks ?